sqlmesh/core/engine_adapter/base.py

```diff
@@ -407,6 +408,20 @@ def create_index(
         )
         self.execute(expression)
 
+    class _KeyForType(Enum):
+        TABLE = "TABLE_TYPE"
+        VIEW = "VIEW_TYPE"
+
+    def _parse_type_out_of_properties(
+        self, properties: t.Dict[str, exp.Expression], type_key: _KeyForType
+    ) -> t.Optional[exp.Expression]:
+        """Parses out the type from the properties dictionary and returns it if it exists. It does also remove the key"""
+        for key in list(properties.keys()):
+            upper_key = key.upper()
+            if upper_key == type_key.value:
+                return properties.pop(key)
+        return None
```

Comment:
[Nit] should this be named "extract_creatable_type_from_properties"? The verb "parse" is a misnomer.
```suggestion
        return next(properties.pop(key) for key in properties if key.upper() = type_key.value, None)
```

Reply:
Avoided the commit as introduced a little more inline complexity.